### PR TITLE
Add starter code to repository

### PR DESCRIPTION
**Pull request by borjasotomayor**
Pull request to merge started code into repository (see #2)

### CODE REVIEW

File: src/libobj/src/parser.c [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+int myPrint(char* string){
+    if (DEBUG){
+        printf("%s\n",string);
+    }
+    return 0;
+}
+
+/* See parser.h */
+stack_t* begin_sequence(stack_t* stack, char* key){
+    stackobj_t* stacktop = stack -> top;
```
Comment (mpcsh) [inline src/libobj/src/parser.c]: Arrows shouldn't have spaces around them

File: src/libobj/examples/parse.c [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#include <stdio.h>
+#include "parser.h"
+
+int main(int argc, char* argv[]){
```
[the remark this answers is not in the document]
Comment (dwahme) [reply src/libobj/examples/parse.c]: Done. We're sticking to allman style

File: src/libobj/include/obj.h [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+ *  - id: The id of the attribute to get the value of
+ * 
+ * Returns:
+ *  - Returns the value in the object if successfully found
+ *  - Returns the 0 equivalent for that datatype if not found or 
+ *      wrong type
+ */
+bool obj_get_bool(obj_t *obj, char *id);
+char obj_get_char(obj_t *obj, char *id);
+int   obj_get_int(obj_t *obj, char *id);
```
[the remark this answers is not in the document]
Comment (dwahme) [reply src/libobj/include/obj.h]: Done

File: src/libobj/include/obj.h [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+ *  - id: The id of the attribute to set the value of
+ *  - value: The value to set the attribute's value to
+ * 
+ * Returns:
+ *  - Returns EXIT_SUCCESS if successfully set
+ *  - Returns EXIT_FAILURE on failure
+ */
+int obj_set_bool(obj_t *obj, char *id, bool  value);
+int obj_set_char(obj_t *obj, char *id, char  value);
+int  obj_set_int(obj_t *obj, char *id, int   value);
```
[the remark this answers is not in the document]
Comment (dwahme) [reply src/libobj/include/obj.h]: Done. Not quite like that beautiful functional look anymore

File: src/libobj/src/obj.c [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+
+#include "obj.h"
+#include "utlist.h"
+
+/* Forward function declarations */
+obj_t *obj_get_attr_single(obj_t *obj, char *id, bool create);
+void obj_free_str(obj_t *obj);
+
+/* See obj.h */
+obj_t *obj_new(char *id) {
```
[the remark this answers is not in the document]
Comment (dwahme) [reply src/libobj/src/obj.c]: Done

File: src/libobj/src/parser.c [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+#include "parser.h"
+
+/* DEBUG is 0 normally, 1 to print debugging statements */
+#define DEBUG 0
+
+/* Forward function declaration */
+void itoa(int n, char s[]);
+
+/* Helper function: to print debugging statments only when debugging */
+int myPrint(char* string){
```
[the remark this answers is not in the document]
Comment (dwahme) [reply src/libobj/src/parser.c]: Done

File: src/libobj/src/parser.c [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+    if (DEBUG){
+        printf("%s\n",string);
+    }
+    return 0;
+}
+
+/* See parser.h */
+stack_t* begin_sequence(stack_t* stack, char* key){
```
[the remark this answers is not in the document]
Comment (dwahme) [reply src/libobj/src/parser.c]: Done

File: src/libobj/src/parser.c (hunk 1 above)
Comment (dwahme) [reply src/libobj/src/parser.c]: Done

**Comment by dwahme** [comment]
All requested changes have been completed @mpcsh @lydiafilipe

**Comment by mpcsh** [comment]
@borjasotomayor I think the master branch protection might be blocking us here lmao

**Comment by borjasotomayor** [comment]
@mpcsh: I can actually override this as an administrator. In the future, I expect pull requests that originate in one of us three (Lydia, you, and me) get reviewed by the other two, plus Dawson xor Maxine xor Firat (however, in this case, we were merging code written by all of them so...)

**Pull request merged**